### PR TITLE
Add `thrust_create_target` `DISPATCH` option.

### PR DESCRIPTION
**Pull request by alliepiper**
## Description

Add `DISPATCH` option to `thrust_create_target`.
    
```
thrust_create_target(TargetName
  DISPATCH [Dynamic|Force32bit|Force64bit]
)
```

Adds 32/64-bit dispatch jobs to nightly CI. Updated docs. Various drive-by fixes.

CI Jobs with 32-/64-bit forced dispatch: https://github.com/NVIDIA/cccl/actions/runs/11900778160/job/33162677687

## Checklist

- [X] New or existing tests cover these changes.
- [X] The documentation is up to date with these changes.

### CODE REVIEW

File: lib/cmake/thrust/README.md [hunk 1]
```diff
@@ -38,6 +38,22 @@ If using Thrust from the CCCL sources, this would be
...
 ```
 
+#### Large Array (64-bit offseet) Handling: `DISPATCH`
+
+The `DISPATCH` option allows users to select the tradeoff of compile-time / binary-size
+vs. performance vs. scalability when given large inputs that require 64-bit offset types.
+This currently only applies when DEVICE=CUDA.
+
+- `Dynamic` Compiles each kernel twice, once for 32-bit offsets and again for 64-bit offsets,
+  and chooses dynamically using the input size at runtime.
```
Comment (bernhardmgruber) [inline lib/cmake/thrust/README.md:48]: Suggestion: IIUC, this is not always true. Some algorithms switched to int64 regardless and others just chunk the input into 32-bit addressable pieces. Maybe rephrase to `May compile each kernel twice`.

File: thrust/testing/unique_by_key.cu [hunk 2]
```diff
@@ -448,6 +450,8 @@ struct TestUniqueCopyByKeyLargeOutCount
 };
 SimpleUnitTest<TestUniqueCopyByKeyLargeOutCount, IntegralTypes> TestUniqueCopyByKeyLargeOutCountInstance;
 
+#  endif // force 32-bit dispatch
```
Comment (bernhardmgruber) [inline thrust/testing/unique_by_key.cu:453]:
```suggestion
#  endif // THRUST_FORCE_32_BIT_OFFSET_TYPE
```

**Review by bernhardmgruber — APPROVED** [review]

File: thrust/testing/cuda/partition.cu [hunk 3]
```diff
@@ -677,9 +677,12 @@ void TestPartitionIfWithMagnitude(int magnitude)
 void TestPartitionIfWithLargeNumberOfItems()
 {
   TestPartitionIfWithMagnitude(30);
+  // These require 64-bit dispatches even when magnitude < 32.
+#  ifndef THRUST_FORCE_32_BIT_OFFSET_TYPE
```
Comment (bdice) [inline thrust/testing/cuda/partition.cu:681]: Do we typically indent macros? Most in this PR are not indented, but there are a few exceptions.

**Review by bdice — COMMENTED** [review]
Thanks for working on this! Will this target 2.7 or 2.8? We can always backport this on the RAPIDS side if it solves the problem. I will try to test it with cuDF today.

File: thrust/testing/cuda/partition.cu (hunk 3 above)
Comment (alliepiper) [reply thrust/testing/cuda/partition.cu:681]: Our clang-format did that.

Comment (alliepiper) [reply thrust/testing/cuda/partition.cu:681]: Note that it isn't indenting to the surrounding code level, but because this `#ifndef` is inside of another `#if`.

**Comment by alliepiper** [comment]
I believe this will go into 2.8.0.

File: thrust/testing/cuda/partition.cu (hunk 3 above)
Comment (bdice) [reply thrust/testing/cuda/partition.cu:681]: Gotcha, thanks for the explanation!
